include/ddc/kernels/splines/matrix.hpp

```diff
@@ -107,11 +170,23 @@ class Matrix
         return bx;
     }
 
+    /**
+     * @brief Get the size of the matrix (which is necessarily squared) in one of its dimensions.
+     *
+     * @return The size of the matrix in one of its dimensions.
+     */
     int get_size() const
     {
         return m_n;
     }
 
+    /**
+     * @brief Prints a Matrix in a std::ostream. It will segfault is the Matrix is on GPU.
+     *
+     * @param out The stream in which the matrix is printed.
+     *
+     * @return The stream in which the matrix is printed.
+     **/
     std::ostream& operator<<(std::ostream& os)
```

Comment:
Do you know if this function works and if so how to use it? When this code was in Gysela the `operator<<` was defined outside the class and I was able to write `cout << m` with `m` a Matrix. I tried this in DDC last week and couldn't get it to work. I think that's because class methods are approximately equivalent to external functions whose first argument is an instance of the class (`std::ostream& operator<<(Matrix const& m, std::ostream& os)`) but this is not the expected prototype for the `operator<<`. 

I guess now it must be something like:
```cpp
std::cout << "My matrix:";
m << std::cout;
```
?

Reply:
No, I think I never used it. You think it should be removed ?

Reply:
I think it should be moved back to where it was (https://gitlab.maisondelasimulation.fr/gysela-developpers/gyselalibxx/-/blob/8ae3867a42d91e7713705d1123044d6fcd2855b2/vendor/sll/include/sll/matrix.hpp) so it can continue to be used for debugging purposes

Reply:
#439